### PR TITLE
Media: Dispatch validation error redux actions from flux actions

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR:

* Introduces new media validation actions where necessary wrt validation errors
* Changes validation store so it dispatches validation redux actions when the corresponding flux actions are dispatched.

#### Testing instructions

* Checkout this branch.
* Verify all tests pass.
* Try smoke testing media library and verify it still works well.

Note: This PR is part of an iteration to remove the media validation store, see #39372 for the full effort that removes it completely.

### CODE REVIEW

File: client/lib/media/actions.js [hunk 1]
```diff
@@ -189,6 +209,10 @@ function uploadFiles( uploader, files, site ) {
 							data: data.media[ 0 ],
 						} )
 					);
+
+					reduxDispatch( successMediaItemRequest( siteId, transientMedia.ID ) );
```
Comment (jameslnewell) [inline client/lib/media/actions.js]: Does this need to be two distinct actions? It feels a bit [like the actions are `setters`](https://redux.js.org/style-guide/style-guide/#model-actions-as-events-not-setters).

Comment (tyxla) [reply client/lib/media/actions.js]: Yeah, in this case, it's preferred for keeping things simple - `receiveMedia` will handle multiple items, while `successMediaItemRequest` is for the request of a single media item. 

I agree with you these could be combined into a single action, but note that these action creators were there before creating this PR, so I'm just using them wherever it makes sense. 

That being said, we could always come back and refactor here when everything is using Redux.

File: client/state/media/actions.js [hunk 2]
```diff
@@ -80,13 +84,15 @@ export function requestingMedia( siteId, query ) {
  *
  * @param  {number} siteId Site ID
  * @param  {object} query  Query object
+ * @param  {object} error  Error object
  * @returns {object}        Action object
  */
-export function failMediaRequest( siteId, query ) {
+export function failMediaRequest( siteId, query, error = null ) {
```
Comment (blowery) [inline client/state/media/actions.js]: why null vs leaving it `undefined` if it's not set?

**Review by blowery — APPROVED** [review]
This LGTM. Adding an error prop to the failMediaRequest action appears to be unchecked, so it should be safe, and all the other new actions are new and unhandled by reducers or middleware.

File: client/state/media/actions.js (hunk 2 above)
Comment (tyxla) [reply client/state/media/actions.js]: Because once we start using those errors in #39375 dealing with an undefined was throwing an error.

**Comment by tyxla** [comment]
Going to keep the default error prop value as it's [necessary](https://github.com/Automattic/wp-calypso/pull/39374#discussion_r388477810) in the subsequent PRs.

**Comment by tyxla** [comment]
Thanks @blowery!

**Pull request merged**